packages/grafana-ui/src/components/Card/Card.tsx

```diff
@@ -21,83 +18,181 @@ export interface Props extends Omit<CardContainerProps, 'disableEvents' | 'disab
-      href={href}
-      {...htmlProps}
-    >
-      {figure}
-      <div className={styles.inner}>
-        <div className={styles.info}>
-          <div>
-            <h2 className={styles.heading}>{heading}</h2>
-            {meta}
-            {description && <p className={styles.description}>{description}</p>}
-          </div>
-          {tags}
-        </div>
-        {hasActions && (
-          <div className={styles.actionRow}>
-            {actions}
-            {secondaryActions}
-          </div>
-        )}
-      </div>
+    <CardContainer disableEvents={disabled} disableHover={disableHover} {...htmlProps}>
+      {!hasHeading && <Heading href={href} onClick={onCardClick}></Heading>}
+      {subComponents}
     </CardContainer>
   );
 };
 
-function getEnterKeyHandler(onClick: () => void) {
-  return (e: React.KeyboardEvent<HTMLInputElement>) => {
-    if (e.key === 'Enter') {
-      onClick();
+interface ChildProps {
+  className?: string;
+  styles?: ReturnType<typeof getCardStyles>;
+  disabled?: boolean;
+}
+
+const Heading: FC<ChildProps & { onClick?: () => void; href?: string; 'aria-label'?: string }> = ({
+  children,
+  href,
+  onClick,
+  styles,
+  className,
+  'aria-label': ariaLabel,
+}) => {
+  return (
+    <h2 className={cx(styles?.heading, className)}>
+      {href ? (
+        <a href={href} className={styles?.linkHack} aria-label={ariaLabel}>
+          {children}
+        </a>
+      ) : onClick ? (
+        <button className={styles?.linkHack} onClick={onClick} aria-label={ariaLabel}>
+          {children}
+        </button>
+      ) : (
+        <>{children}</>
+      )}
+    </h2>
+  );
+};
+Heading.displayName = 'Heading';
+
+const Tags: FC<ChildProps> = ({ children, styles, className }) => {
+  return <div className={cx(styles?.tagList, className)}>{children}</div>;
+};
+Tags.displayName = 'Tags';
+
+const Description: FC<ChildProps> = ({ children, styles, className }) => {
+  return <p className={cx(styles?.description, className)}>{children}</p>;
+};
+Description.displayName = 'Description';
+
+const Figure: FC<ChildProps & { align?: 'start' | 'center' }> = ({ children, styles, align = 'start', className }) => {
+  return (
+    <div
+      className={cx(
+        styles?.media,
+        className,
+        css`
+          align-self: ${align};
+        `
+      )}
+    >
+      {children}
+    </div>
+  );
+};
+Figure.displayName = 'Figure';
+
+const Meta: FC<ChildProps & { separator?: string }> = memo(({ children, styles, className, separator = '|' }) => {
+  let meta = children;
+
+  // Join meta data elements by separator
+  if (Array.isArray(children) && separator) {
+    const filtered = React.Children.toArray(children).filter(Boolean);
+    if (!filtered.length) {
+      return null;
     }
-  };
+    meta = filtered.reduce((prev, curr, i) => [
+      prev,
+      <span key={`separator_${i}`} className={styles?.separator}>
+        {separator}
+      </span>,
+      curr,
+    ]);
+  }
```

Comment:
Imagine if we had a component that was like 

```
<Interpose node={separator}> 
  <A />
  <B />
</Interpose>
```

that could do this for us 😍 

(not a suggestion for now, just a mild thought for later 😄  )